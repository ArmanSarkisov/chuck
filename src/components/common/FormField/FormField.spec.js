import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { FormField } from './index';

describe('FormField test', () => {
  let component;
  let mockCallback;
  let mockObject;

  beforeEach(() => {
    mockCallback = jest.fn();
    mockObject = { mock: true };
    component = shallow(
      <FormField
        message="message"
        register={mockCallback}
        control={mockObject}
        placeholder="placeholder"
        type="text"
        text="some text"
        inputId="id"
      />,
    );
  });

  it('should render Card component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render props register', () => {
    expect(mockCallback.mock.calls.length).toBe(0);
  });
});
