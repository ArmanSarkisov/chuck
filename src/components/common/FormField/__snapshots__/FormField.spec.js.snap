// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField test should render Card component 1`] = `
<div
  className="mb-4"
>
  <Label
    error="message"
    htmlFor="id"
    text="some text"
  />
  <Input
    control={
      Object {
        "mock": true,
      }
    }
    id="id"
    placeholder="placeholder"
    register={[MockFunction]}
    type="text"
  />
</div>
`;
